util/dnsname/dnsname.go

```diff
@@ -17,3 +96,21 @@ func HasSuffix(name, suffix string) bool {
 	nameBase := strings.TrimSuffix(name, suffix)
 	return len(nameBase) < len(name) && strings.HasSuffix(nameBase, ".")
 }
+
+// SanitizeHostname returns hostname with some common suffixes removed.
```

Comment:
that's not all it does, though

Reply:
I might instead just call this TrimCommonSuffixes and don't do the SanitizeLabel part, punting that to the caller.

Reply:
👍 I updated to break this out into two functions: `TrimCommonSuffixes` and `SanitizeHostname`.

Kept around `SanitizeHostname` so we can simplify
```
hostname := dnsname.TrimCommonSuffixes(node.Hostname)
hostname = dnsname.SanitizeLabel(hostname)
```
to just
```
hostname := dnsname.SanitizeHostname(node.Hostname)
```
since it's used in a few spots.